docs/user_guide/02b_query_engine.md

```diff
@@ -1 +1,294 @@
+setup-lines: 1-4
+lines: 27-43
+---
+```
+
+:::{dropdown} Methods used in the snippet
+
+- [`in_group()`](medmodels.medrecord.querying.NodeOperand.in_group){target="_blank"} : Query nodes that belong to that group.
+- [`index()`](medmodels.medrecord.querying.NodeOperand.index){target="_blank"}: Returns a [`NodeIndexOperand`](medmodels.medrecord.querying.NodeIndexOperand){target="_blank"}` to query on the indices.
+- [`contains()`](medmodels.medrecord.querying.NodeIndexOperand.contains){target="_blank"} : Query node indices containing that argument.
+- [`attribute()`](medmodels.medrecord.querying.NodeOperand.attribute){target="_blank"} : Returns a [`MultipleValuesOperand`](medmodels.medrecord.querying.MultipleValuesOperand){target="_blank"} to query on the values of the nodes for that attribute.
+- [`lowercase()`](medmodels.medrecord.querying.MultipleValuesOperand.lowercase){target="_blank"} : Converts the values that are strings to lowercase.
+- [`trim()`](medmodels.medrecord.querying.MultipleValuesOperand.trim){target="_blank"} : Removes leading and trailing whitespacing from the values.
+- [`equal_to()`](medmodels.medrecord.querying.MultipleValuesOperand.equal_to){target="_blank"} : Query values equal to that value.
+
+- [`has_attribute()`](medmodels.medrecord.querying.NodeOperand.has_attribute){target="_blank"} : Query nodes that have that attribute.
+- [`mean()`](medmodels.medrecord.querying.MultipleValuesOperand.mean){target="_blank"}: Returns a [`SingleValueOperand`](medmodels.medrecord.querying.SingleValueOperand){target="_blank"} containing the mean of those values.
+- [`subtract()`](medmodels.medrecord.querying.SingleValueOperand.subtract){target="_blank"} : Subtract the argument from the single value operand.
+
+- [`less_than()`](medmodels.medrecord.querying.MultipleValuesOperand.less_than){target="_blank"} : Query values that are less than that value.
+- [`select_nodes()`](medmodels.medrecord.medrecord.MedRecord.select_nodes){target="_blank"} : Select nodes that match that query.
+
+:::
+
+:::{note}
+Query methods used for changing the operands cannot be concatenated or assigned to variables, since their `Return` is null. That is, the following code snippet will set `gender_lowercase` as `None`, and thus the query will not be able to find any nodes.
+
+```python
+# Wrong implementation
+gender_lowercase = node.attribute("gender").lowercase()
+gender.equal_to("m")
+
+# Wrong implementation
+gender = node.attribute("gender")
+gender.lowercase().trim()
+gender.equal_to("m")
+
+# Correct implementation
+gender = node.attribute("gender")
+gender.lowercase()
+gender.trim()
+gender.equal_to("m")
+```
+
+Nor do the ones that compare operands to other operands, since their `Return` value is also null.
+
+```python
+# Wrong implementation
+gender = node.attribute("gender")
+gender.equal_to("M").not_equal_to("F")
+
+# Correct implementation
+gender = node.attribute("gender")
+gender.equal_to("M")
+gender.not_equal_to("F")
+```
+
+:::
```

Comment:
very important part, noice!

---

the return value is `None` and not `null`. I also don't think I'd put return (or Return) in the `code comment mode`

---

did you mean
```python
# Wrong implementation
gender_lowercase = node.attribute("gender").lowercase()
gender_lowercase.equal_to("m")
```
?

---

I also wouldn't say that

> [...] the query will not be able to find any nodes.

as these queries would result in a `AttributeError` to be thrown, e.g.
```python
AttributeError("'NoneType' object has no attribute 'equal_to'")
```